docs/uploading.md

```diff
@@ -15,16 +15,18 @@ Bind a [file input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/in
 {{/let}}
 ```
 
-After a file is added to your queue, the `onFileAdded` callback will be called. You may then upload the file to a URL by calling `file.upload()`. This returns a promise that is resolved when the file has finished uploading, or is rejected if the file couldn't be uploaded.
+After a file is added to your queue, the `onFileAdded` callback will be called – for a complete callback reference see the [file-queue Helper API](../docs/api/helper/file-queue.md#helper-api).
+
+You may start to upload the file by calling `file.upload()`. This returns a promise that is resolved when the file has finished uploading, or is rejected if the file couldn't be uploaded.
 
 ```js
 export default class ExampleComponent extends Component {
   @action
   async uploadPhoto(file) {
     try {
       const response = await file.upload('/api/images/upload');
-    } catch (error) {
-      file.state = 'aborted';
+    } catch (response) {
+      // Handle error response somehow
     }
   }
 }
```

Comment:
@jelhan as you say I don't think there's a reason consumers should update state. So I'm suggesting we replace this example

Reply:
I feel that make sense. Maybe we could give a more realistic example? I guess informing the user and providing an option to retry the upload might be most common error handling?

Something like this:

```js
export default class ExampleComponent extends Component {
  @action
  async uploadPhoto(file) {
    try {
      const response = await file.upload('/api/images/upload');
    } catch (response) {
      if (!response.ok) {
        const shouldRestart = window.confirm(
          'Uploading failed due to broken network connectivity. Please check your internet connection. Do you want to retry?'
        );
        
        if (shouldRestart) {
          await this.uploadPhoto(file);
        }
      } else {
        window.alert('Upload failed with a server-side error. Please try again later.');
      }
    }
  }
}
```

Or do we want to push the people towards `onUploadFailed` for error handling?

Reply:
I think an example like this is good starting point but I'm very conscious that many users will copy and paste example code directly into their app (I've seen it happen).

Given this, we'd want a well-considered example that's going to be best-practice for most users.

I think advanced usage like retries, s3 uploads etc could probably have their own "Recipes" pages on the doc site. We could pull some example code from old issue solutions, too.